packages/core/src/conflict-resolution.ts

```diff
@@ -260,34 +294,24 @@ export class ConflictResolution {
       const genesis = await this.dispatcher.retrieveCommit(genesisCid)
       await handler.applyCommit(genesis, { cid: genesisCid, timestamp: timestamp }, this.context)
     }
-    const itr = unappliedCommits.entries()
-    let entry = itr.next()
-    while (!entry.done) {
-      const logEntry = await this.getCommitData(entry.value[1])
-      const commitMeta = { cid: logEntry.cid, timestamp: logEntry.timestamp }
-      // TODO - should catch potential thrown error here
-
-      if (StreamUtils.isAnchorCommit(logEntry.commit)) {
-        // It's an anchor commit
-        // TODO: Anchor validation should be done by the StreamHandler as part of applying the anchor commit
-        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, logEntry.commit)
-        state = await handler.applyCommit(logEntry.commit, commitMeta, this.context, state)
-      } else {
-        // It's a signed commit but may not be using DagJWS. If the JWS envelope is present, use that while applying the
-        // commit, otherwise use the commit itself (e.g. for CAIP-10 links).
-        const commitToApply = logEntry.envelope ? logEntry.envelope : logEntry.commit
-        const tmpState = await handler.applyCommit(commitToApply, commitMeta, this.context, state)
-        const isGenesis = !logEntry.commit.prev
-        const effectiveState = isGenesis ? tmpState : tmpState.next
-        // TODO: Schema validation should be done by the StreamHandler as part of applying the commit
-        await this.stateValidation.validate(effectiveState, effectiveState.content)
-        state = tmpState // if validation is successful
+
+    for (const entry of unappliedCommits) {
+      try {
+        state = await this.applyLogEntryToState(entry, state, handler)
+      } catch(err) {
+        // TODO(1586): include StreamID in log message
+        this.context.loggerProvider.getDiagnosticsLogger().warn(
+          `Error while applying commit ${entry.cid.toString()}: ${err}`)
+        if (opts.throwOnApplyCommitError) {
```

Comment:
I'm open to other names for this parameter.  `requireSuccessfulCommitApplication` maybe?

Reply:
Vote for `throwOnInvalidCommit`, but this is a weak opinion. Like `throw` prefix.

Reply:
I like @ukstv suggestion